### ISSUE TITLE
Incorrect evaluation results

### ISSUE THREAD

**Issue by XiHuYan** [AI4SCR/MatchCLOT#2]
Hi, I found tha [run.py](https://github.com/AI4SCR/MatchCLOT/blob/main/matchclot/run/run.py) produced very low evaluation scores. It was caused by incorrect ground truth of match solutions. The hyperparameter `CUSTOM_DATASET_PATH` was `""` by default but you used `if CUSTOM_DATASET_PATH is False` for judgement in [run.py](https://github.com/AI4SCR/MatchCLOT/blob/32d8b3e0622d2f1ab634bbb12d9664f38507bf76/matchclot/run/run.py#LL319C42-L319C42), leading to wrong matching solutions. Maybe you should add some notes about the hyperparameter settings :)

**Comment by XiHuYan** [comment]
By the way, I found the code for calculating FOSCTTM score is wrong. Line 76 [evaluate.py](https://github.com/AI4SCR/MatchCLOT/blob/32d8b3e0622d2f1ab634bbb12d9664f38507bf76/matchclot/run/evaluate.py#L76):
`foscttm_x = (X > torch.diag(X)).float().mean(axis=1).mean().item()`
should be 
`foscttm_x = (X > torch.diag(X).view(-1, 1)).float().mean(axis=1).mean().item()`

**Comment by XiHuYan** [comment]
I found that your definition of **Fraction of Samples Closer Than the True Match (FOSCTTM)** differs from the original study [1](https://www.biorxiv.org/content/10.1101/2020.06.13.149195v1.abstract). They directly used the Eculidean distance between cells to compute FOSCTTM scores but you used the predicted matching matrix to compute the scores, which is equivalent to adding a transformation layer on the distance matrix.  Obviously, your calculation process can strengthen the FOSCTTM scores. Besides, CLUE's definition of FOSCTTM score is the same as the original study, which means that your comparison with CLUE in terms of FOSCTTM scores is unfair.

**Comment by FedericoGossi** [comment]
> Hi, I found tha [run.py](https://github.com/AI4SCR/MatchCLOT/blob/main/matchclot/run/run.py) produced very low evaluation scores. It was caused by incorrect ground truth of match solutions. The hyperparameter `CUSTOM_DATASET_PATH` was `""` by default but you used `if CUSTOM_DATASET_PATH is False` for judgement in [run.py](https://github.com/AI4SCR/MatchCLOT/blob/32d8b3e0622d2f1ab634bbb12d9664f38507bf76/matchclot/run/run.py#LL319C42-L319C42), leading to wrong matching solutions. Maybe you should add some notes about the hyperparameter settings :)

Hello! Thank you for bringing this issue to our attention, we have fixed it with commit 34356191531d6d7ae9e04cdeef8768bfef722cb9. In general, if you use a custom dataset by setting CUSTOM_DATASET_PATH, the automatic evaluation in run.py expects that the two modalities have the cell profiles in the same order. You can follow the [quickstart tutorial](https://matchclot.readthedocs.io/en/latest/source/quickstart.html) to run MatchCLOT on an example dataset. If you are willing to reproduce the results of our paper, we suggest to follow the instructions and the code in the [paper-reproducibility](https://github.com/AI4SCR/MatchCLOT/tree/paper-reproducibility) branch. There, the default configuration should run on the NeurIPS competition dataset and CUSTOM_DATASET_PATH can be left to the default "".

**Comment by XiHuYan** [comment]
Thanks for your reply.

I cannot access to the download link of pretrained model in the [paper-reproducibility](https://github.com/AI4SCR/MatchCLOT/tree/paper-reproducibility) branch.

**Comment by FedericoGossi** [comment]
> I found that your definition of **Fraction of Samples Closer Than the True Match (FOSCTTM)** differs from the original study [1](https://www.biorxiv.org/content/10.1101/2020.06.13.149195v1.abstract). They directly used the Eculidean distance between cells to compute FOSCTTM scores but you used the predicted matching matrix to compute the scores, which is equivalent to adding a transformation layer on the distance matrix. Obviously, your calculation process can strengthen the FOSCTTM scores. Besides, CLUE's definition of FOSCTTM score is the same as the original study, which means that your comparison with CLUE in terms of FOSCTTM scores is unfair.

Thank you for bringing up this concern.
Firstly, I want to emphasize that the evaluation of MatchCLOT against CLUE was not solely based on the FOSCTTM scores. The competition metric was also used in the comparison, and MatchCLOT consistently showed superior performance.
Secondly, it's important to clarify that the FOSCTTM is a metric specifically designed to evaluate the matching quality of a model, rather than the quality of the embeddings themselves. While it's true that our calculation process involves using the predicted matching matrix to compute the FOSCTTM scores, this approach is intended to assess the effectiveness of the matching task performed by the model. Since the OT matching algorithm in MatchCLOT plays a fundamental role in its performance, as demonstrated in the paper results, it would not be fair to compare MatchCLOT to other models solely based on the evaluation of the contrastive learning embeddings.
We appreciate your feedback, if you have any further questions or suggestions, please don't hesitate to let us know.
